courses/machine_learning/deepdive/08_image/mnist_linear.ipynb

```diff
@@ -9,9 +9,7 @@
    "source": [
```

Comment:
<p>This is interesting. What's the advantage to use that instead of example code using the function with assert (which would be simpler to read and convey more easily the information on how to use the code to the reader as well as a mean to test it)? I am a bit afraid with that that the testing section grows as big as the notebook itself if we use the full power of the unittest suite. What do you think?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/training-data-analyst/pull/551/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
Yeah, after thinking about it, if we can pull the code from the notebook to run in automated tests, asserts would be easier to work with, huh? One advantage I like from the unit test suite is it gives a nice print out explicitly stating what passed and didn't pass. I switched to asserts and add a "Success!" print out if it passes, so hopefully best of both worlds.